FI/src/Functional.java

```diff
@@ -0,0 +1,131 @@
+                if (s.length() % 2 == 0) return s.substring(0, s.length() / 2);
+                return "";
+            };
+            System.out.println(unary.apply("abcd"));
+            System.out.println(unary.apply("abcde"));
+
+//        8. Return a random number
+            Supplier<Double> supplier1 = () -> Math.random();
+            System.out.println(supplier1.get());
+
+            Supplier<Integer> supplier2 = () -> {
+                Double d = Math.random() * 100;
+                return d.intValue();
+            };
+            System.out.println(supplier2.get());
+
+//        9. Given a string, return a reversed string: "abc" ->"cba"
+            UnaryOperator<String> unary1 = s -> {
+                StringBuilder sb = new StringBuilder(s);
+                sb = sb.reverse();
+                return String.valueOf(sb);
+            };
+            System.out.println(unary1.apply("abc"));
+
+//        10. Given a number, return a reversed number: 12345 -> 54321
+            UnaryOperator<Integer> unary2 = u -> {
+                int i = 0;
+                while (u > 0) {
+                    i += u % 10;
+                    i *= 10;
+                    u = u / 10;
+                }
+                return i / 10;
+            };
+            System.out.println(unary2.apply(12345));
+
+//        11. Given a number, return true if it is even, false otherwise: 6 -> true, 25 -> false
+            Predicate<Integer> p1 = num -> num % 2 == 0;
+            System.out.println(p1.test(6));
+            System.out.println(p1.test(25));
+
+//        12. Given a string, return true, if its length is more than 3, false otherwise: "ab" -> false, "abc" -> true
+            // ??? "is more than 3" or "and gleich"?)
+            Predicate<String> p2 = s -> s.length() >= 3;
+            System.out.println(p2.test("ab"));
+            System.out.println(p2.test("abc"));
+
+//        13. Given a string and a number x. Return true, if length of string is more than x, false otherwise. Hint: BiPredicate.
+            BiPredicate<String, Integer> bp = (string_A, b) -> string_A.length() > b;
+
+//    Print a given string
+            Consumer<String> consumer = System.out::println;
+            consumer.accept("string");
+
+//    Given a string, return a lower case string
+            UnaryOperator<String> unaryString = String::toLowerCase;
+            System.out.println(unaryString.apply("STRING"));
+
+//    Given a number, return a square root from the number
+            UnaryOperator<Double> un2 = Math::sqrt;
+
+//    Implement solution 7 from above as a method reference
+//        String is not a functional interface
+
+//    Given a class Employee with name and salary. Return an employee salary using 1. Supplier, 2. Function.
+//        Supplier<Employee> supplier = ()->{
+//            return new Employee("Vasja", 100);
+//        };
+            Supplier<Employee> supplier = Employee::new;
```

Comment:
This supplier returns an object of class Employee. The idea was that f we have an Employee object already, you can return its salary using a supplier on an instance:
`Supplier<Integer> getSalary = employee::getSalary;`
where `employee` is an object of class `Employee`.